packages/block-library/src/list/edit.js

```diff
@@ -178,6 +178,7 @@ export default function ListEdit( {
 				start={ start }
 				reversed={ reversed }
 				type={ type }
+				{ ...useBlockProps() }
```

Comment:
`useBlockProps` is a React hook, shouldn't it be at the top of the component here and it other places to ensure it is always predictable (according to [rules of hooks](https://en.reactjs.org/docs/hooks-rules.html))?

Shouldn't those spread props be applied as the first item? In the current shape, if there is any clash in naming, block wrapper props will erase previously listed props.

Reply:
I think it's fine to inline if possible. We should also try to avoid early returns in components if hooks are being used. It's pointless to execute the hook otherwise.

Reply:
I'm with @gziolo here. I don't think hooks should be inlined in this way. It's fine technically, but it seems like it'd be very easy for devs that might be changing this code later to miss the presence of this hook.

Reply:
We have the `react-hooks/rules-of-hooks` ESLint rule configured, no? That should catch any mistakes due to using the hook in a branch.

Reply:
Yep, I tested locally and `react-hooks/rule-of-hooks` will complain if you return before using `useBlockProps`.

```
gutenberg (try/use-block *) % git diff
diff --git a/packages/block-library/src/list/edit.js b/packages/block-library/src/list/edit.js
index a5094cb569..e283bc3b3a 100644
--- a/packages/block-library/src/list/edit.js
+++ b/packages/block-library/src/list/edit.js
@@ -51,6 +51,10 @@ export default function ListEdit( {
                return !! select( 'core/block-editor' ).getSettings().isRTL;
        }, [] );
 
+       if ( isRTL ) {
+               return null;
+       }
+
        const controls = ( { value, onChange, onFocus } ) => (
                <>
                        { isSelected && (
gutenberg (try/use-block *) % npm run lint-js packages/block-library/src/list/edit.js

> gutenberg@8.4.0-rc.1 lint-js /Users/robert/Code/gutenberg
> wp-scripts lint-js "packages/block-library/src/list/edit.js"


/Users/robert/Code/gutenberg/packages/block-library/src/list/edit.js
  185:10  error  React Hook "useBlockProps" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks

✖ 1 problem (1 error, 0 warnings)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gutenberg@8.4.0-rc.1 lint-js: `wp-scripts lint-js "packages/block-library/src/list/edit.js"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gutenberg@8.4.0-rc.1 lint-js script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/robert/.npm/_logs/2020-08-04T02_32_25_789Z-debug.log
```